### PR TITLE
add filter pareto front from eval util

### PR DESCRIPTION
**Pull request by LGro**
The time has come and we benefit from a function that just takes a list of evaluations and filters out the pareto front, in addition to / building on the previous implementation of the loss matrix based mask.

Signed-off-by: Grossberger Lukas (CR/AIR2.2) <Lukas.Grossberger@de.bosch.com>

### CODE REVIEW

File: tests/utils_test.py [hunk 1]
```diff
@@ -42,3 +46,46 @@ def test_get_loss_vector_with_custom_none_replacement():
...
+        Evaluation(configuration={"i": 4}, objectives={"loss": 3.1, "score": -1.1}),
+        Evaluation(configuration={"i": 5}, objectives={"loss": 0.1, "score": -1.0}),
+        Evaluation(configuration={"i": 6}, objectives={"loss": 0.0, "score": -1.1}),
+        Evaluation(configuration={"i": 7}, objectives={"loss": -1.0, "score": -2.0}),
+    ]
+    pareto_efficient = filter_pareto_efficient(
+        evals, [Objective("loss", False), Objective("score", True)]
+    )
+    assert len(pareto_efficient) == 4
+    assert set([0, 2, 3, 7]) == set([e.configuration["i"] for e in pareto_efficient])
```
Comment (LGro) [inline tests/utils_test.py:91]: This is now duplicating test logic from `mask_pareto_efficient` more than it would need. Effectively, we just need to test here that the objective inversion and filtering of the list work. However, I found the overhead low enough and the benefit large enough that we get many edge cases here on this level as well to accept the duplication. Let me know what you think about this aspect.

File: blackboxopt/visualizations/utils.py [hunk 2]
```diff
@@ -172,27 +175,6 @@ def add_plotly_buttons_for_logscale(fig):
...
-    for i, c in enumerate(costs):
-        if not is_efficient[i]:
-            continue
-
-        # Keep any point with a lower cost or when they are the same
-        efficient = np.any(costs[is_efficient] < c, axis=1)
-        duplicates = np.all(costs[is_efficient] == c, axis=1)
-        is_efficient[is_efficient] = np.logical_or(efficient, duplicates)
-
-    return is_efficient
```
Comment (LGro) [inline blackboxopt/visualizations/utils.py:193]: This just moved to the general, visualization unspecific, utils (so did the test).

File: tests/utils_test.py (hunk 1 above)
Comment (dynobo) [reply tests/utils_test.py:91]: Agree, it's worth it!
